src/NuGet.Clients/NuGet.VisualStudio.Common/ServiceProviderExtensions.cs

```diff
@@ -60,7 +60,15 @@ public static IComponentModel GetComponentModel(this IServiceProvider servicePro
             this Microsoft.VisualStudio.Shell.IAsyncServiceProvider site)
             where TInterface : class
         {
-            return await site.GetServiceAsync(typeof(TService)) as TInterface;
+            var service = site.GetServiceAsync(typeof(TService));
```

Comment:
should we expect 'await' here?

Reply:
Also, you want to ConfigureAwait(false), so if GetServiceAsync does get the service on the UI thread (for non-free threaded services), hopefully, it is still in the UI thread.  That will make the line 67 can finish immediately, and reduce the overhead of jumping between two threads.

Reply:
nice catch, yes forgot to add `await` there. And thanks for the suggestion to add `ConfigureAwait(false)`

Reply:
I disagree with `.ConfigureAwait(false)`. You will *usually* get the service you're asking for, so what this suffix would do is require yet one more thread transition. Instead of resuming directly on the main thread, you'll resume on a threadpool thread then have to switch to the main thread. Why not just leave the suffix off so that you optimize the case when your caller was already on the main thread?

Reply:
@AArnott aren't we already using `.ConfigureAwait(false)` for `ServiceProvider.GetGlobalServiceAsync()` as well?

Reply:
@jainaashish It looks like when your everted the `.ConfigureAwait(false)` suffix, you also reverted your fix to @lifengl's first comment (to add a missing `await` to your `GetServiceAsync` call.

> aren't we already using .ConfigureAwait(false) for ServiceProvider.GetGlobalServiceAsync() as well?

Yes, but only in the overload that _doesn't_ have a subsequent cast in it. In the one that does have a cast, we leave the suffix off.